authui/src/authflowv2/customSelect.ts

```diff
@@ -212,11 +212,11 @@ export class CustomSelectController extends Controller {
   }
 
   handleSelect(event: MouseEvent) {
-    const item = event.target as HTMLLIElement | undefined;
-    if (!item) return;
-
-    const value = item.dataset.value;
-    this._selectValue(value);
+    const item = event.target;
+    if (item instanceof HTMLLIElement) {
+      const value = item.dataset.value;
+      this._selectValue(value);
+    }
   }
```

Comment:
This might be incorrect: It could be other type of element inside the li element if we use event.target. (Like the span inside it)
- Maybe we should use currentTarget, and checking instanceof HTMLLIElement will be redundant because it is always the element we attached the listener to.

Reply:
You mean the original code is correct? I just rewrite it in a way that preserve the original meaning.

Reply:
I've left a new comment about the type assertion error:
https://github.com/authgear/authgear-server/pull/5272#discussion_r2192113520

For this code, I feel like the original code should use currentTarget.